### PR TITLE
Add SITL documentation

### PR DESCRIPTION
**Pull request by haslinghuis**

### CODE REVIEW

**Comment by haslinghuis** [comment]
@freasy 

did not remove Building in Windows document
added a SITL document with /img and /resource items

Somehow automation is making it hard to contribute as it did work locally in test browser

**Comment by freasy** [comment]
> @freasy
> 
> did not remove Building in Windows document added a SITL document with /img and /resource items
> 
> Somehow automation is making it hard to contribute as it did work locally in test browser

The build is fine, only name and title checks fail, which are not that important. Where should I look at?

File: docs/development/SITL.mdx [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# SITL
+
+## SITL in Gazebo
+
+SITL (software in the loop) simulator allows you to run betaflight without any hardware.
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:5]:
```suggestion
SITL (software in the loop) simulator allows you to run Betaflight without any hardware.
```

File: docs/development/SITL.mdx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ArduPilot also offers [RealFlight SITL](https://ardupilot.org/dev/docs/sitl-with-realflight.html).
+To use it you may need to purchase it on [Steam](https://store.steampowered.com/app/1070820/RealFlight_95S/) or [offical website](https://www.realflight.com/).
+
+### Setup
+
+To let BetaFlight SITL work with RealFlight, you need Windows 10 or Windows 11 with WSL.
+Ubuntu 20.04 in WSL2 on Windows 11 x64 is tested.
+[RealFlightBridge](https://github.com/xuhao1/RealFlightBridge) is also required.
+
+On WSL2, you need to configure the BetaFlight following [document here](/docs/development/Building-in-Windows).
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:62]:
```suggestion
On WSL2, you need to configure the Betaflight following [document here](/docs/development/Building-in-Windows).
```

File: docs/development/SITL.mdx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+Build BetaFlight with
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:64]:
```suggestion
Build Betaflight with
```

File: docs/development/SITL.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+$ make TARGET=SITL
+```
+
+On Windows, download RealFlightBridge by
+
+```bash
+$ git clone https://github.com/xuhao1/RealFlightBridge.git
+```
+
+Import the quadcopter for RealFlight and BetaFlight at **models/Quadcopter X Betaflight - flightaxis_AV.RFX** in RealFlightBridge. A detailed guide for improtanting can be found [here](https://ardupilot.org/dev/docs/sitl-with-realflight.html).
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:76]:
```suggestion
Import the quadcopter for RealFlight and Betaflight at **models/Quadcopter X Betaflight - flightaxis_AV.RFX** in RealFlightBridge. A detailed guide for improtanting can be found [here](https://ardupilot.org/dev/docs/sitl-with-realflight.html).
```

File: docs/development/SITL.mdx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+2. Starting the RealFlightBridge in **Windows** by
+
+   ```bash
+   $ python betaflight_bridge.py
+   ```
+
+   You must start RealFlightBridge in Windows **before** start the betaflight SITL.
+
+3. Start BetaFlight SITL.
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:101]:
```suggestion
3. Start the Betaflight SITL.
```

File: docs/development/SITL.mdx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+       ether 00:15:5d:a4:21:5d  txqueuelen 1000  (Ethernet)
+       RX packets 219079  bytes 32440158 (32.4 MB)
+       RX errors 0  dropped 0  overruns 0  frame 0
+       TX packets 145744  bytes 10533796 (10.5 MB)
+       TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+   ```
+
+   172.19.32.1 is the example output of your Windows IP and 172.19.41.192 is your WSL IP. **These IPs change everytime you reboot the computer.**
+
+   Then go to the BetaFlight root and start SITL with this IP.
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:121]:
```suggestion
   Then go to the Betaflight root and start the SITL with this IP.
```

File: docs/development/SITL.mdx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+   start UDP server for RC input @9004...0
+   [FLASH_Unlock] loaded 'eeprom.bin', size = 32768 / 32768
+   [timer]Init...
+   [data]new fdm 136 t:182.834571 from 0.0.0.0:0
+   [data]new rc 40: t:182.834571 AETR: 1498 1501 1105 1501 AUX1-4: 1100 1899 1899 1100
+   bind port 5761 for UART1
+   unusedPinsInit
+   ```
+
+   Then you can open BetaFlight configurator on **Windows** to connect to RealFlight via **WSL IP**.
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:140]:
```suggestion
   Then you can open the Betaflight Configurator on **Windows** to connect to RealFlight via **WSL IP**.
```

File: docs/development/SITL.mdx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+   ![](/img/sitl/betaflight.jpg)
+
+   After connect to BetaFlight you need to apply this [DIFF file](/sitl/BTFL_quadcopter_rf9.txt) and set the arming switch with your controller.
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:143]:
```suggestion
   After connect to Betaflight you need to apply this [DIFF file](/sitl/BTFL_quadcopter_rf9.txt) and set the arming switch with your controller.
```

File: docs/development/SITL.mdx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+4. Finally, you can arm and take-off the Quadcopter with BetaFlight SITL in RealFlight with your controller.
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:145]:
```suggestion
4. Finally, you can arm and take-off the Quadcopter with Betaflight SITL in RealFlight with your controller.
```

File: docs/development/SITL.mdx [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+In `configuration` page:
+
+1. `ESC/Motor`: `PWM`, disable `Motor PWM speed Sparted from PID speed`
+2. `PID loop frequency` as high as it can.
+
+### Start and Run
+
+1. start betaflight: `./obj/main/betaflight_SITL.elf`
+2. start gazebo: `gazebo --verbose ./iris_arducopter_demo.world`
+3. connect your transmitter and fly/test, used app to send `MSP_SET_RAW_RC`, see [code](https://github.com/cs8425/msp-controller).
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:36]:
```suggestion
1. Start Betaflight: `./obj/main/betaflight_SITL.elf`
2. Start Gazebo: `gazebo --verbose ./iris_arducopter_demo.world`
3. Connect your transmitter and fly/test, used app to send `MSP_SET_RAW_RC`, see [code](https://github.com/cs8425/msp-controller).
```

File: docs/development/SITL.mdx [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+:::note
+
+betaflight -> gazebo `udp://127.0.0.1:9002`
+gazebo -> betaflight `udp://127.0.0.1:9003`
+
+UARTx will bind on `tcp://127.0.0.1:576x` when port been open.
+
+`eeprom.bin`, size 8192 Byte, is for config saving.
+size can be changed in `src/main/target/SITL/pg.ld` >> `__FLASH_CONFIG_Size`
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:46]:
```suggestion
Size can be changed in `src/main/target/SITL/pg.ld` >> `__FLASH_CONFIG_Size`
```

File: docs/development/SITL.mdx [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+### Start and Run
+
+1. start betaflight: `./obj/main/betaflight_SITL.elf`
+2. start gazebo: `gazebo --verbose ./iris_arducopter_demo.world`
+3. connect your transmitter and fly/test, used app to send `MSP_SET_RAW_RC`, see [code](https://github.com/cs8425/msp-controller).
+
+:::note
+
+betaflight -> gazebo `udp://127.0.0.1:9002`
+gazebo -> betaflight `udp://127.0.0.1:9003`
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:41]:
```suggestion
Betaflight -> Gazebo `udp://127.0.0.1:9002`
Gazebo -> Betaflight `udp://127.0.0.1:9003`
```

File: docs/development/SITL.mdx [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+```bash
+$ git clone https://github.com/xuhao1/RealFlightBridge.git
+```
+
+Import the quadcopter for RealFlight and BetaFlight at **models/Quadcopter X Betaflight - flightaxis_AV.RFX** in RealFlightBridge. A detailed guide for improtanting can be found [here](https://ardupilot.org/dev/docs/sitl-with-realflight.html).
+Moreover, update the setting of RealFlight to allow API.
+![](/img/sitl/rf_settings.jpg)
+
+You also need to prepare a controller for running SITL.
+I recommend to use transmitter with OpenTX/EdgeTX as a game controller. In addition, mixers of channel 5 and 6 should be mapped to two switches for arming and changing mode.
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:81]:
```suggestion
We recommend to use transmitter with OpenTX/EdgeTX as a game controller. In addition, mixers of channel 5 and 6 should be mapped to two switches for arming and changing mode.
```

File: docs/development/SITL.mdx [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+
+   ```bash
+   $ python betaflight_bridge.py
+   ```
+
+   You must start RealFlightBridge in Windows **before** start the betaflight SITL.
+
+3. Start BetaFlight SITL.
+   First you need to get the Windows' IP in WSL.
+   In WSL entering
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:103]:
```suggestion
   In WSL enter
```

**Review by KarateBrot — CHANGES_REQUESTED** [review]

**Comment by haslinghuis** [comment]
Done. Thanks @KarateBrot.

File: docs/development/Building-in-Windows.md [hunk 15]
```diff
@@ -0,0 +1,15 @@
+# Building in Windows
+
+For various reasons, a custom build of Betaflight might be needed. A user might want to enable certain features that are disabled on his / her target, for example. This guide provides the basic instructions needed to build Betaflight on a Windows system. One options is given:
```
Comment (KarateBrot) [inline docs/development/Building-in-Windows.md:3]:
```suggestion
For various reasons, a custom build of Betaflight might be needed. A user might want to enable certain features that are disabled on his / her target, for example. This guide provides the basic instructions needed to build Betaflight on a Windows system. One option is given:
```

File: docs/development/SITL.mdx [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+
+https://ubuntu.com/blog/install-gazebo-for-ros-2-in-under-a-minute
+
+:::
+
+Please also check [Vehicle Gateway](https://github.com/osrf/vehicle_gateway)
+
+### Build Betaflight
+
+run `make TARGET=SITL`
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:21]:
```suggestion
Run `make TARGET=SITL`
```

File: docs/development/SITL.mdx [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+
+### Settings
+
+to avoid simulation speed slow down, suggest to set some settings belows:
```
Comment (KarateBrot) [inline docs/development/SITL.mdx:25]:
```suggestion
To avoid the simulation speed slowing down, it is suggested to set some settings shown below:
```

**Review by KarateBrot — CHANGES_REQUESTED** [review]
Found some more 😀

Btw. do we need to worry about the failing github checks?

**Review by freasy — APPROVED** [review]

**Review by KarateBrot — APPROVED** [review]

**Pull request merged**